src/main/java/com/monumental/listeners/MonumentListener.java

```diff
@@ -0,0 +1,52 @@
+package com.monumental.listeners;
+
+import com.monumental.models.Monument;
+import com.monumental.services.MonumentService;
+import org.hibernate.event.spi.PreInsertEvent;
+import org.hibernate.event.spi.PreInsertEventListener;
+import org.hibernate.event.spi.PreUpdateEvent;
+import org.hibernate.event.spi.PreUpdateEventListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MonumentListener implements PreInsertEventListener, PreUpdateEventListener {
+
+    @Autowired
+    MonumentService monumentService;
+
+    /**
+     * Before a Monument is inserted, generate its slug
+     */
+    @Override
+    public boolean onPreInsert(PreInsertEvent event) {
+        // This event is called for all inserts so filter out non-Monuments
+        if (!(event.getEntity() instanceof Monument)) {
+            return false;
+        }
+
+        Monument monument = (Monument) event.getEntity();
+        monumentService.generateSlug(monument);
+
+        return false;
```

Comment:
Why return false if we successfully generated the slug?

Reply:
For some reason Hibernate uses this return value to check if the operation should be aborted
https://docs.jboss.org/hibernate/orm/3.5/api/org/hibernate/event/PreInsertEventListener.html

> Return true if the operation should be vetoed